src/targets/gpu/gemm_impl.cpp

```diff
@@ -574,17 +574,45 @@ void gemm_compute(context& ctx,
     gemm_item.run(ctx, args, solution_idx);
 }
 
+static value gemm_problem(const shape& output_shape, std::vector<shape> input_shapes)
+{
+    input_shapes.push_back(output_shape);
+    return to_value(input_shapes);
+}
+
+static void gemm_save_solution(context& ctx,
+                               const shape& output_shape,
+                               const std::vector<shape>& input_shapes,
+                               int32_t solution_idx)
+{
+    ctx.get_problem_cache().insert(
+        "rocblas", gemm_problem(output_shape, input_shapes), solution_idx);
+}
+
+int32_t gemm_default_solution(context& ctx,
+                              const shape& output_shape,
+                              const std::vector<shape>& input_shapes)
+{
+    std::vector<shape> gemm_shapes{input_shapes[0], input_shapes[1], output_shape};
+    auto problem = to_value(gemm_shapes);
```

Comment:
This intended to be fed into gemm_problem as the shapes?

Reply:
What do you mean?

Reply:
the varaible problem isn't used but I assume captures shapes and would be used in place of gemm_problem(output, input)?

Reply:
Ah thats a mistake. Good catch. I guess cppcheck caught it as well. The code just needs to be deleted, as I replaced it with a function since the problem needs to be computed in more than one place.

Reply:
All good. LGTM after fix.